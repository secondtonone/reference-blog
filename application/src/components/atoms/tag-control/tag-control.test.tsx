import renderer from 'react-test-renderer';
import Link from 'next/link';
import TagControl from '.';

jest.mock('next/link', () => jest.fn().mockReturnValue('link'));

describe('Match snapshot TagControl', () => {
  it('with button', () => {
    const tree = renderer.create(<TagControl />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with link', () => {
    const tree = renderer.create(<TagControl href="/" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(Link).toHaveBeenCalled();
  });
});
