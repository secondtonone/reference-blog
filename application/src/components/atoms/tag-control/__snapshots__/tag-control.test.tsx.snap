// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot TagControl with button 1`] = `
<span
  className="text-content__TextStyled-sc-1stkrsz-0 gnSSuA tag-control__TagControlStyled-sc-5ljs9a-0 fuGmRv"
  data-test="tag-control"
>
  <button
    type="button"
  />
</span>
`;

exports[`Match snapshot TagControl with link 1`] = `
<span
  className="text-content__TextStyled-sc-1stkrsz-0 gnSSuA tag-control__TagControlStyled-sc-5ljs9a-0 fuGmRv"
  data-test="tag-control"
>
  link
</span>
`;
